backend/tests/unit/test_twitter_extract_text.py

```diff
@@ -0,0 +1,297 @@
+        api_response = {
+            'data': {
+                'tweetResult': {
+                    'result': {
+                        'legacy': {
+                            'full_text': '급처분 포카 양도합니다 15,000원'
+                        }
+                    }
+                }
+            }
+        }
+
+        tweet_text = self.service._extract_tweet_text(api_response)
+
+        self.assertEqual(tweet_text, '급처분 포카 양도합니다 15,000원')
+
+    def test_extract_tweet_text_from_visibility_results(self):
+        """Test extracting tweet text from TweetWithVisibilityResults structure."""
+        api_response = {
+            'data': {
+                'tweetResult': {
+                    'result': {
+                        'tweet': {
+                            'legacy': {
+                                'full_text': '급처분 양도'
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        tweet_text = self.service._extract_tweet_text(api_response)
+
+        self.assertEqual(tweet_text, '급처분 양도')
+
+    def test_extract_tweet_text_missing_data(self):
+        """Test that ValueError is raised when tweet data is missing."""
+        api_response = {'data': {}}
+
+        with self.assertRaises(ValueError) as context:
+            self.service._extract_tweet_text(api_response)
+
+        self.assertIn("Failed to parse tweet content", str(context.exception))
+
+    def test_extract_tweet_text_invalid_structure(self):
+        """Test that ValueError is raised for invalid response structure."""
+        api_response = {'invalid': 'structure'}
+
+        with self.assertRaises(ValueError) as context:
+            self.service._extract_tweet_text(api_response)
+
+        self.assertIn("Failed to parse tweet content", str(context.exception))
+
+    # ==============================================
+    # Integration Tests with Mocked API
+    # ==============================================
+
+    @patch('trade_safety.twitter_extract_text_service.requests.get')
+    def test_fetch_tweet_content_success(self, mock_get):
+        """Test fetching tweet content with mocked API response."""
+        # Given: Twitter URL
+        url = "https://x.com/mkticket7/status/2000111727493718384"
+
+        # Mock API response with note_tweet structure
+        mock_response = MagicMock()
+        mock_response.json.return_value = {
+            'data': {
+                'tweetResult': {
+                    'result': {
+                        'tweet': {
+                            'note_tweet': {
+                                'note_tweet_results': {
+                                    'result': {
+                                        'text': '2025 권진아 연말 콘서트 티켓양도\n\n12/24\n중앙블럭 1열 연석 14.5'
+                                    }
+                                }
+                            }
+                        }
+                    }
```

Comment:
현재 구현이 Twitter API v2 (`data.text`)를 사용하는데, mock 응답이 구버전 구조(`note_tweet`)라서 테스트가 실패합니다.


```suggestion
                'text': '2025 권진아 연말 콘서트 티켓양도\n\n12/24\n중앙블럭 1열 연석 14.5'
```